### DISCUSSION TITLE
How can I change the maximum allowed setpoint?

### DISCUSSION THREAD

**Discussion by eecharlie** [ScratMan/HASmartThermostat#256] [Q&A]
I'm trying to use this controller for a likely off-label application, drying filament for a 3D printer. I need to be able to use set points as high as 140F or so. How can I change this enforced limit on the GUI? 

(I'm happy to edit code, I'm just a Python guy and Home Assistant feels like a labyrinth of files that might contain what I'm looking for)

**Comment by Chupaka** [comment]
> this enforced limit

What limit do you mean? 

```
    min_temp: 0
    max_temp: 200
```
These parameters should allow you to do what you want

**Reply by eecharlie** [reply]
Thanks, maybe I am generally confused about how to use semi-customized integrations in HA - it seems like manual yaml editing exists in a parallel universe to GUI-configured things.
- If I create a new integration with the GUI workflow, there is no opportunity to set `max_temp`, and I cannot find one in GUI menus after it's created. Additionally, I cannot find any yaml file anywhere containing the `climate: ` block.
- If I add the example code to my `configuration.yaml`, it has no effect. No pid-related entities exist and I can't find anything to add to my dashboard, even after restarting. I also get the error "Platform error 'climate' from integration 'smart_thermostat' - Integration 'smart_thermostat' not found."

What am I missing here?

**Reply by Chupaka** [reply]
> If I create a new integration with the GUI workflow

Is there one?.. For me, it says "This device cannot be added from the UI. You can add this device by adding it to your 'configuration.yaml'. See the documentation for more information."

**Reply by eecharlie** [reply]
Ah, thanks, oof, I was getting confused because I had a *different* self-tuning temperature PID controller integration installed. So now I've got it nominally working, and I've turned on the suggested debug variables, but the control gains are all coming up zero. 

What might explain that, and how can I get the pretty plot showing duty cycle shading etc that I see on the project readme?

**Reply by Chupaka** [reply]
> how can I get the pretty plot showing duty cycle shading etc that I see on the project readme?

Click three dots at top right corner of thermostat panel, then History icon at top right corner
<img width="298" alt="image" src="https://github.com/user-attachments/assets/f755de6d-1ffc-416a-8dfb-ac49da6e783f">
The plot you'll see can help to explain what happens
